src/native.rs

```diff
@@ -21,16 +24,33 @@ impl Context {
         F: FnMut(&str) -> *const std::os::raw::c_void,
     {
         let raw = native_gl::Gl::load_with(loader_function);
-        let constants = Constants {
-            max_label_length: {
-                let mut val = 0;
-                unsafe {
-                    raw.GetIntegerv(MAX_LABEL_LENGTH, &mut val);
-                }
-                val
-            },
+
+        // Setup extensions and constants after the context has been built
+        let mut context = Self {
+            raw,
+            extensions: HashSet::new(),
+            constants: Constants::default(),
+        };
+
+        // Use core-only functions to populate extension list
+        // TODO: Use a fallback for versions < 3.0
+        use crate::Context;
```

Comment:
would be nicer to move it at least to the beginning of the function and import `as _` to stress that it's just a trait